.github/workflows/os_test.yml

```diff
@@ -2,45 +2,13 @@ name: course_os test suite
 
 on: [push]
 
-defaults:
-  run:
-     working-directory: src
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
 
-      - name: Cache Toolchain
-        id: cache-toolchain
-        uses: actions/cache@v1
-        with:
-            path: src/toolchain
-            key: toolchain
-
-      - name: Cache Qemu
-        id: cache-qemu
-        uses: actions/cache@v1
-        with:
-            path: src/qemu/qemu
-            key: qemu
-
-      - name: Install Dependencies
-        run: |
-          sudo apt-get update &&
-          sudo apt-get install build-essential python3 wget texinfo zlib1g-dev ninja-build -y
-
-      - name: Build Toolchain
-        if: steps.cache-toolchain.outputs.cache-hit != 'true'
-        run: rm -rf toolchain/arm-none-eabi/ && make toolchain
-
-      - name: Build Qemu
-        if: steps.cache-qemu.outputs.cache-hit != 'true'
-        run: rm -rf qemu/qemu* && make qemu
-
-      - name: Compile
-        run: make build
-
-      - name: Test
-        run: make test
+      - run: |
```

Comment:
Maybe it's good to keep build and test different stages. In that way we can 
1. see easier what's failing, building or tests
2. if we want to ever export build artifacts that's a bit saner

Reply:
oh I didn't mean seperating the docker build stage (though that's also a good idea), but building the kernel itself.

Reply:
Docker will build the kernel when you build the image so it can export the volume with the kernel, but you could change it

Reply:
oh alright, misremembered that then